packages/kbn-build/src/utils/packages.js

```diff
@@ -0,0 +1,137 @@
+import _glob from 'glob';
+import path from 'path';
+import promisify from 'pify';
+
+import { CliError } from './errors';
+import { Package } from './Package';
+
+const glob = promisify(_glob);
+
+export async function getPackages(rootPath, packagesPaths) {
+  const globOpts = {
+    cwd: rootPath,
+    strict: true,
+    absolute: true
+  };
+  const packages = new Map();
+
+  for (const globPath of packagesPaths) {
+    const res = await glob(path.join(globPath, 'package.json'), globOpts);
+
+    for (const globResult of res) {
+      const packageConfigPath = normalize(globResult);
+      const packageDir = path.dirname(packageConfigPath);
+      const pkg = await Package.fromPath(packageDir);
+
+      packages.set(pkg.name, pkg);
+    }
+  }
+
+  return packages;
+}
+
+// https://github.com/isaacs/node-glob/blob/master/common.js#L104
+// glob always returns "\\" as "/" in windows, so everyone
+// gets normalized because we can't have nice things.
+function normalize(dir) {
+  return path.normalize(dir);
+}
+
+function buildPackageGraph(packages) {
+  const packageGraph = new Map();
+
+  for (const pkg of packages.values()) {
+    const packageDeps = [];
+    const dependencies = pkg.allDependencies;
+
+    for (const depName of Object.keys(dependencies)) {
+      const depVersion = dependencies[depName];
+
+      if (packages.has(depName)) {
+        const dep = packages.get(depName);
+
+        pkg.ensureValidPackageVersion(dep, depVersion);
```

Comment:
__note:__ maybe I'm missing something, but this piece of code feels a bit weird :)
* The name `ensureValidPackageVersion` gives an impression that it can accept any package as argument, but in fact it expects only dependencies of this package. Maybe rename it to `ensureValidDependencyVersion`?
* `pkg` already has access to the `depVersion` internally so there is no need to pass it as argument, it needs only `dep.name` (to extract `depVersion` from its internal collection of dependencies or throw if it doesn't have such dependency) and `dep.path`. What about `ensureValidDependencyVersion(depName, depVersion)` signature instead?